src/boundaries.rs

```diff
@@ -102,7 +102,55 @@ pub fn build_boundary(
     relation: &osmpbfreader::Relation,
     objects: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>,
 ) -> Option<MultiPolygon<f64>> {
-    let roles = ["outer", "enclave", ""];
+    use geo::algorithm::contains::Contains;
+
+    let outer_polys = build_boundary_parts(relation, objects, vec!["outer", "enclave", ""]);
+    let inner_polys = build_boundary_parts(relation, objects, vec!["inner"]);
+    let mut multipoly = MultiPolygon(vec![]);
+
+    if inner_polys.is_none() {
+        return outer_polys;
+    }
+    let mut inner_polys = inner_polys.unwrap();
+
+    if outer_polys.is_some() {
+        let outer_polys = outer_polys.unwrap();
+        for outer in outer_polys.0 {
+            let mut added_outer = false;
+            let mut i = 0;
+            let mut inners = vec![];
+            while i != inner_polys.0.len() {
+                let current_inner = inner_polys.0.get(i).unwrap().clone();
+                if outer.contains(&current_inner) {
+                    inner_polys.0.remove(i);
+                    inners.push(current_inner.exterior);
+                    added_outer = true;
+                } else {
+                    i += 1;
+                };
+            }
+            if !added_outer {
+                multipoly.0.push(outer.clone());
+            } else {
+                multipoly
```

Comment:
I don't know if that would'nt be too costly to cpoy those big geometry around, it might be better to change add the inner into the outer